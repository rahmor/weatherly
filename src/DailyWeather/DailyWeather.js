import React from 'react';
import './DailyWeather.css';

function DailyWeather() {
  return (
    <div className='DailyWeather-container'>
      <div className='DailyWeather-temp'>
        <h3>Atl, Ga Weather</h3>
        <p>as of 10:00pm</p>
        <p>91</p>
        <p>Partly Cloudy</p>
      </div>
      <div className='DailyWeather-icon'>
        <h3>Weather Icon</h3>
        <p>Hi-93, Lo-75%</p>
      </div>
      <div></div>
    </div>
  );
}

export default DailyWeather;
