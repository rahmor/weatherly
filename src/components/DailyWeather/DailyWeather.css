.DailyWeather-temp,
.DailyWeather-icon {
  box-shadow: 0 0 10px purple;
  border-radius: 2px;
  width: 120px;
  margin: auto;
}
