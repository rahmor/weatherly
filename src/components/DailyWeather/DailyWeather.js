import React from 'react';
import './DailyWeather.css';

export class DailyWeather extends React.Component {
  displayWeather = () => {
    if (this.props.fetchError) {
      return <h1>Please Enter a Location to get the Weather </h1>;
    } else {
      return (
        <>
          <h3>{this.props.city} Weather</h3>
          <p>as of {this.props.current.time}</p>
          <p>{this.props.current.temp}</p>
          <p>{this.props.current.condition}</p>{' '}
        </>
      );
    }
  };

  render() {
    return (
      <div className='DailyWeather-container'>
        <div className='DailyWeather-temp'>
          {this.props.fetching ? <h1>Loading</h1> : this.displayWeather()}
        </div>
        <div className='DailyWeather-icon'>
          {this.props.fetching ? (
            <h1>Loading</h1>
          ) : (
            <>
              <h3>Weather Icon</h3>
              <p>
                Hi-{this.props.current.hi}, Lo-{this.props.current.lo}
              </p>
            </>
          )}
        </div>
      </div>
    );
  }
}

export default DailyWeather;
