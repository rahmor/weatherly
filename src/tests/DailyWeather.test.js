import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DailyWeather from '../components/DailyWeather/DailyWeather';

describe('DailyWeather component', () => {
  it('should render UI as expected', () => {
    const wrapper = shallow(<DailyWeather />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
