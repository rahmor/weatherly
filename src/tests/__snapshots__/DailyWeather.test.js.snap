// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeather component should render UI as expected 1`] = `
<div
  className="DailyWeather-container"
>
  <div
    className="DailyWeather-temp"
  >
    <h3>
       Weather
    </h3>
    <p>
      as of 10:00pm
    </p>
    <p>
      91
    </p>
    <p>
      Partly Cloudy
    </p>
  </div>
  <div
    className="DailyWeather-icon"
  >
    <h3>
      Weather Icon
    </h3>
    <p>
      Hi-93, Lo-75%
    </p>
  </div>
  <div />
</div>
`;
